freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/cleanup/CleanupService.java

```diff
@@ -315,6 +325,36 @@ private void removeHostRelatedServices(FreeIpaClient client, Set<String> hostsTo
         return Pair.of(vaultCleanupSuccess, vaultCleanupFailed);
     }
 
+    private void waitForIpaServerDeletion(Long stackId, FreeIpaClient client, Set<String> hosts) throws FreeIpaClientException {
+        // The IPA server delete continues running after the API returns. The last step is for FreeIPA to delete the topology segments.
+        for (int i = 0; i < serverDeletionCheckRetries; i++) {
```

Comment:
Done, I think there are other places in the code that probably should do the same, for example, `FreeIpaPollerService` and many other `PollerServices`. I fixed this one but didn't cleanup everything else.

Reply:
I think there are 2 accepted ways of polling in CB code:
- using the `PollingService`
- using `com.dyngr.polling`

I'm fine with both of them. I would like to avoid another `for` cycle with `Thread#sleep` in the code. I proposed the first one as I used that one most of the time, but feel free to use the other one as well if you like it better.